src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -55,6 +55,18 @@ const TopicFragment = ({
       following: item.following !== undefined ? item.following : item.user_id_follower !== null
     }));
   }, [discovery.initialTopics]);
+  const newMapFollowedTopics = React.useMemo(() => {
+    return followedTopic.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [followedTopic]);
+  const newMapUnfollowedTopics = React.useMemo(() => {
+    return unfollowedTopic.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [unfollowedTopic]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useMemo` hooks for `newMapFollowedTopics` and `newMapUnfollowedTopics` seem to be doing the same operation on different arrays. This is a violation of the DRY (Don't Repeat Yourself) principle. Consider creating a helper function that performs this operation and can be used in both hooks.
```diff
-  const newMapFollowedTopics = React.useMemo(() => {
-    return followedTopic.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
-  }, [followedTopic]);
-  const newMapUnfollowedTopics = React.useMemo(() => {
-    return unfollowedTopic.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
-  }, [unfollowedTopic]);

+  const mapFollowingStatus = (topics) => {
+    return topics.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  };
+
+  const newMapFollowedTopics = React.useMemo(() => mapFollowingStatus(followedTopic), [followedTopic]);
+  const newMapUnfollowedTopics = React.useMemo(() => mapFollowingStatus(unfollowedTopic), [unfollowedTopic]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->